tests/Argentina/ArgentinaTest.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+/*
+ * This file is part of the Yasumi package.
+ *
+ * Copyright (c) 2015 - 2021 AzuyaLabs
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @author Sacha Telgenhof <me@sachatelgenhof.com>
+ */
+
+namespace Yasumi\tests\Argentina;
+
+use ReflectionException;
+use Yasumi\Holiday;
+use Yasumi\tests\ProviderTestCase;
+
+/**
+ * Class for testing holidays in Argentina.
+ */
+class ArgentinaTest extends ArgentinaBaseTestCase implements ProviderTestCase
+{
+    /**
+     * @var int year random year number used for all tests in this Test Case
+     */
+    protected $year;
+
+    /**
+     * Initial setup of this Test Case.
+     */
+    protected function setUp(): void
+    {
+        $this->year = $this->generateRandomYear(1980);
+    }
+
+    /**
+     * Tests if all official holidays in Argentina are defined by the provider class.
+     *
+     * @throws ReflectionException
+     */
+    public function testOfficialHolidays(): void
+    {
+        $this->assertDefinedHolidays([
+            'newYearsDay',
+            'internationalWorkersDay',
+            'remembranceDay',
+            'malvinasDay',
+            'mayRevolution',
+            'generalMartinMigueldeGuemesDay',
+            'flagDay',
+            'independenceDay',
+            'generalJoseSanMartinDay',
+            'raceDay',
+            'nationalSovereigntyDay',
+            'immaculateConceptionDay',
+            'christmasDay',
+        ], self::REGION, $this->year, Holiday::TYPE_OFFICIAL);
+    }
+
+    /**
+     * Tests if all observed holidays in Argentina are defined by the provider class.
+     *
+     * @throws ReflectionException
+     */
+    public function testObservedHolidays(): void
+    {
+        $this->assertDefinedHolidays([
```

Comment:
This asserts that these holidays all are defined from the start test year (1980), which is correct. However it doesn't assert that the respective holidays are not present before their establishment year.

Reply:
Same as above ^^